js/client.js

```diff
@@ -806,12 +813,17 @@
 				var roomTypeLFIndex = roomType.indexOf('\n');
 				if (roomTypeLFIndex >= 0) roomType = roomType.substr(0, roomTypeLFIndex);
 				roomType = toId(roomType);
+
 				if (this.rooms[roomid] || roomid === 'staff') {
 					// staff is always joined in background
 					this.addRoom(roomid, roomType, true);
 				} else {
 					this.joinRoom(roomid, roomType, true);
 				}
+
+				if (roomType === 'chat' && roomid !== 'staff') {
+					this.addAutojoin(roomid, true);
```

Comment:
Rerunning `sliceAutojoin` might be more reliable than manually doing `addAutojoin` and `removeAutojoin` each time. It's also more likely to be in the right order.